### PR TITLE
[Feature] Mixpanel Subspec

### PR DESCRIPTION
**Pull request by yoyoyoseob**

### CODE REVIEW

File: Simcoe/Adobe/Adobe.swift [hunk 1]
```diff
@@ -47,20 +47,36 @@ extension Adobe: EventTracking {
...
     ///
     /// - Parameters:
-    ///   - amount: The amount to increase that lifetime value for.
-    ///   - item: The optional item to extend.
-    ///   - properties: The optional additional properties.
+    ///   - property: The property.
+    ///   - value: The amonut to increment the property by.
     /// - Returns: A tracking result.
-    public func increaseLifetimeValue(byAmount amount: Double, forItem item: String?, withAdditionalProperties properties: Properties?) -> TrackingResult {
+    public func increment(property: String?, value: Double, withAdditionalProperties properties: Properties?) -> TrackingResult {
```
Comment (htinlinn) [inline Simcoe/Adobe/Adobe.swift]: Change the name of this function to 
```
increment(property: String?, by value: Double, withAdditionalProperties properties: Properties?)
```

File: Simcoe/Adobe/Adobe.swift [hunk 2]
```diff
@@ -47,20 +47,36 @@ extension Adobe: EventTracking {
...
+    /// Increments the properties.
+    /// All values must be number objects.
+    ///
+    /// - Parameter properties: The properties.
+    /// - Returns: A tracking result.
+    public func increment(properties: Properties, withAdditionalProperties data: Properties?) -> TrackingResult {
+        properties.forEach {
+            var data = data ?? Properties()
+            data[$0.key] = "" as String
+            let value = $0.value as? Double ?? 0
```
Comment (htinlinn) [inline Simcoe/Adobe/Adobe.swift]: Use the previous function instead of doing this again here?

File: Simcoe/Adobe/Adobe.swift [hunk 3]
```diff
@@ -47,20 +47,36 @@ extension Adobe: EventTracking {
...
+        ADBMobile.trackLifetimeValueIncrease(NSDecimalNumber(value: value), data: data)
+        return .success
+    }
+
+    /// Increments the properties.
+    /// All values must be number objects.
+    ///
+    /// - Parameter properties: The properties.
+    /// - Returns: A tracking result.
+    public func increment(properties: Properties, withAdditionalProperties data: Properties?) -> TrackingResult {
```
Comment (htinlinn) [inline Simcoe/Adobe/Adobe.swift]: It's kinda odd that the argument for `withAdditionalProperties` is called `data` here and `properties` in the previous function.

File: Simcoe/Adobe/Adobe.swift (hunk 1 above)
Comment (yoyoyoseob) [reply Simcoe/Adobe/Adobe.swift]: Updated in PR #30

File: Simcoe/Adobe/Adobe.swift (hunk 3 above)
Comment (yoyoyoseob) [reply Simcoe/Adobe/Adobe.swift]: Updated in PR #30

File: Simcoe/Adobe/Adobe.swift (hunk 2 above)
Comment (yoyoyoseob) [reply Simcoe/Adobe/Adobe.swift]: Updated in PR #30

**Comment by yoyoyoseob** [comment]
The branch hierarchy has been refactored to place code changes in relevant sections.

This PR will deal as closely to purely _Mixpanel_ changes while PR #30 will handle all protocol related changes

**Review by atecle — APPROVED** [review]

File: Simcoe/Mixpanel/Mixpanel.swift [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+                instanceName: String = UUID().uuidString) {
+        Mixpanel.initialize(token: token,
+                            launchOptions: launchOptions,
+                            flushInterval: flushInterval,
+                            instanceName: instanceName)
+    }
+
+
+    /// Force uploading of queued data to the Mixpanel server.
+    public func stop() {
```
Comment (htinlinn) [inline Simcoe/Mixpanel/Mixpanel.swift]: Should this still be called `flush`? Calling this function doesn't really stop the Mixpanel engine, right?

**Review by htinlinn — DISMISSED** [review]

File: Simcoe/Mixpanel/Mixpanel.swift (hunk 4 above)
Comment (yoyoyoseob) [reply Simcoe/Mixpanel/Mixpanel.swift]: Updated to be `flush()` in PR #30 

Haven't added back `stop()` as I am not 100% clear what that functionality would be yet

File: Simcoe/Mixpanel/Mixpanel.swift [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+//  Mixpanel.swift
+//  Simcoe
+//
+//  Created by Yoseob Lee on 2/28/17.
+//  Copyright © 2017 Prolific Interactive. All rights reserved.
+//
+
+import Mixpanel
+
+typealias MixpanelProperties = [String: MixpanelType]
```
Comment (MichaelCampbell) [inline Simcoe/Mixpanel/Mixpanel.swift]: document typealias

**Review by htinlinn — APPROVED** [review]

**Pull request merged**